devices/rtx/scene/surface/material/sampler/CompressedImage2D.cpp

```diff
@@ -0,0 +1,194 @@
+  } else if (m_format == "BC2") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed2;
+  } else if (m_format == "BC2_SRGB") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed2SRGB;
+  } else if (m_format == "BC3") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed3;
+  } else if (m_format == "BC3_SRGB") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed3SRGB;
+  } else if (m_format == "BC4") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed4;
+  } else if (m_format == "BC4_SNORM") {
+    channelFormatKind = cudaChannelFormatKindSignedBlockCompressed4;
+  } else if (m_format == "BC5") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed5;
+  } else if (m_format == "BC5_SNORM") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed5;
+  } else if (m_format == "BC6H_UFLOAT") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed6H;
+  } else if (m_format == "BC6H_SFLOAT") {
+    channelFormatKind = cudaChannelFormatKindSignedBlockCompressed6H;
+  } else if (m_format == "BC7") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed7;
+  } else if (m_format == "BC7_SRGB") {
+    channelFormatKind = cudaChannelFormatKindUnsignedBlockCompressed7SRGB;
+  } else {
+    reportMessage(ANARI_SEVERITY_WARNING,
+        "invalid texture format encountered in CompressedImage2D sampler (%s)",
+        m_format);
+    return;
+  }
+  makeCudaCompressedTextureArray(
+      m_cuArray, m_size, *m_image.get(), channelFormatKind);
+
+  m_texture = makeCudaCompressedTextureObject(m_cuArray,
+      m_filter,
+      m_wrap1,
+      m_wrap2,
+      "clampToEdge",
+      true,
+      channelFormatKind);
+  m_texels = makeCudaCompressedTextureObject(m_cuArray,
+      "nearest",
+      m_wrap1,
+      m_wrap2,
+      "clampToEdge",
+      false,
+      channelFormatKind);
+
+  upload();
+}
+
+bool CompressedImage2D::isValid() const
+{
+  return m_image;
+}
+
+int CompressedImage2D::numChannels() const
+{
+  if (m_format == "BC1_RGB" || m_format == "BC1_RGBA") {
+    return 4;
+  } else if (m_format == "BC1_RGB_SRGB" || m_format == "BC1_RGBA_SRGB") {
+    return 4;
+  } else if (m_format == "BC2" || m_format == "BC2_SRGB") {
+    return 4;
+  } else if (m_format == "BC3" || m_format == "BC3_SRGB") {
+    return 4;
+  } else if (m_format == "BC4" || m_format == "BC4_SNORM") {
+    return 1;
+  } else if (m_format == "BC5" || m_format == "BC5_SNORM") {
+    return 2;
+  } else if (m_format == "BC6H_UFLOAT" || m_format == "BC6H_SFLOAT") {
+    return 4;
+  } else if (m_format == "BC7" || m_format == "BC7_SRGB") {
+    return 4;
+  } else {
+    return 0;
+  }
+}
+
+SamplerGPUData CompressedImage2D::gpuData() const
+{
+  SamplerGPUData retval = Sampler::gpuData();
+  retval.type = SamplerType::COMPRESSED_TEXTURE2D;
+  retval.compressedImage2D.texobj = m_texture;
```

Comment:
Based on the other note, this could fill `retval.image2D` instead, but keep the `type` as `COMPRESSED_TEXTURE2D`. (?)